Library/Homebrew/shims/mac/super/ruby

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# System Ruby's mkmf on Mojave (10.14) and later require SDKROOT set to work correctly.
+
+RUBY_EXECUTABLE="${0##*/}"
+
+IFS=$'\n'
+for path in $(type -aP "$RUBY_EXECUTABLE")
+do
+  if [[ "$path" != "/usr/bin/$RUBY_EXECUTABLE" && "$path" != "$0" && -f "$path" && -x "$path" ]]
+  then
+    exec "$path" "$@"
+  fi
+done
+unset IFS
```

Comment:
What's this bit doing?

Reply:
To be honest, this is something from [the git shim](https://github.com/Homebrew/brew/blob/0629c29b8eb0a372218a5b9bf6b8c0c746a47725/Library/Homebrew/shims/scm/git#L107-L115).

I think the `IFS` means that it splits on newlines rather any whitespace, and so doesn't break if there's a path with a space in it.

Reply:
Could this be shared logic between the two (with a comment)?

Reply:
Possibly. Running `source` with a relative path is finicky since the SCM shim is symlinked across different directory levels.

Reply:
Or I could just do `$HOMEBREW_PREFIX/whatever`. I'm not entirely sure why the SCM shim does things like `$SCM_DIR/../../../../../bin` and `$SCM_DIR/../../../../bin`, or even why it does both?

Reply:
If there's a comment explaining what it's doing and why they are shared: that seems OK instead.